packages/chevrotain/package.json

```diff
@@ -87,9 +89,11 @@
     "devDependencies": {
         "@types/chai": "^4.0.10",
         "@types/mocha": "^5.0.0",
+        "@types/require-from-string": "^1.2.0",
         "@types/sinon-chai": "^3.2.0",
         "chai": "^4.1.2",
         "coveralls": "^3.0.0",
+        "esm": "^3.2.25",
```

Comment:
where are we using the esm module?

Reply:
I am trying to do unit tests on esm-build, and `nyc` can be used together with `esm` to load `.js` file as es module.

Unfortunately, though the coverage testing succeeds in my environment, it always fails in circleci. (So I think I may have to revert the commits about coverage temporarily)

Reply:
ok

Reply:
Its perfectly fine to have 100% coverage only for the regular CommonJS build.
There are some edge cases of tests under certain engines and handling of generated TypeScript code that reduces coverage by introducing new branches.

We do not need coverage for ESM build, just like we do not have coverage for UMD build.
Just make sure the tests pass there. If there are 1-2 tests that are difficult to enable on ESM
it can also be o.k. to ignore those specifically on esm.

- [Ignoring JSDom tests on browsers](https://github.com/SAP/chevrotain/blob/e807461d61c7511d4bcf66b7d9dbee7626231e5f/packages/chevrotain/test/diagrams/render_spec.ts#L15-L22)
- [Ignoring Coverage in redundant branches TSC creates](https://github.com/SAP/chevrotain/blob/master/packages/chevrotain/scripts/fix-coverage-report.js)

Reply:
Now I've finished the unit testing for esm build